File: CS/PS/Site/programmers/lv1/완주하지못한선수.js
function solution(participant, completion) {
  var answer = "";
  let candidate = new Map();
  //parti 반복해서 돌리고 없으면 키추가, +1
  //있으면 값 +1,
  //comple반복해서 해당 키값 -1씩해서
  //filter로 0인사람 리턴
  for (let i of participant) {
    if (!candidate.has(i)) {
      candidate.set(i, 1);
    } else {
      candidate.set(i, candidate.get(i) + 1);
    }
  }
  for (let i of completion){
    if (candidate.has(i)){
      candidate.set(i, candidate.get(i) - 1);
    }
  }
  let a = Array.from(candidate).filter((v) => v[1] == 1)[0][0];
  return a;
}

let participant = ["leo", "kiki", "eden"];
let completion = ["eden", "kiki"];

console.log(solution(participant, completion));

File: CS/PS/복습/progr_lv1/완주하지못한선수.py
# 16:51 ~ 16:55
import collections
def solution(participant, completion):
    answer = ''
    player = collections.defaultdict()
    for parti in participant:
        if parti not in player:
            player[parti] = 1
        else:
            player[parti] += 1
    for com in completion:
        if com in player:
            player[com] -= 1
    for done in player:
        if player[done] == 1:
            answer = done
    return answer